src/internal.c

```diff
@@ -7445,36 +7472,108 @@ int SendKexDhReply(WOLFSSH* ssh)
             #endif
 #endif /* !defined(WOLFSSH_NO_ECDH) */
             }
-#ifndef WOLFSSH_NO_SABER_LEVEL1_SHA256
-            else if (useSaber) {
-                /* This is a KEM. In this case, I need to encapsulate the
-                 * shared secret. */
+#ifndef WOLFSSH_NO_ECDH_SHA2_NISTP256_KYBER_LEVEL1_SHA256
+            else if (useEccKyber) {
+                /* This is a hybrid KEM. In this case, I need to generate my ECC
+                 * keypair, send the public one, use the private one to generate
+                 * the shared secret, use the post-quantum public key to
+                 * generate and encapsulate the shared secret and send the
+                 * ciphertext. */
                 OQS_KEM* kem = NULL;
+                int primeId;
                 ret = 0;
+            #ifdef WOLFSSH_SMALL_STACK
+                ecc_key *pubKey = NULL, *privKey = NULL;
+                pubKey = (ecc_key*)WMALLOC(sizeof(ecc_key), heap,
+                        DYNTYPE_PUBKEY);
+                privKey = (ecc_key*)WMALLOC(sizeof(ecc_key), heap,
+                        DYNTYPE_PRIVKEY);
+                if (pubKey == NULL || privKey == NULL) {
+                    ret = WS_MEMORY_E;
+                }
+            #else
+                ecc_key pubKey[1];
+                ecc_key privKey[1];
+            #endif
 
                 if (ret == 0) {
-                    kem = OQS_KEM_new(OQS_KEM_alg_saber_lightsaber);
+                    kem = OQS_KEM_new(OQS_KEM_alg_kyber_512);
                     if (kem == NULL) {
                         ret = WS_INVALID_ALGO_ID;
                     }
                 }
 
+                primeId  = wcPrimeForId(ssh->handshake->kexId);
+                if (primeId == ECC_CURVE_INVALID)
```

Comment:
This will overwrite a previous error `(ret != 0)`. Consider also adding check for `if (ret == 0)` here.